components/Modal.tsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import {Modal, View, StyleSheet, Text, TouchableOpacity} from 'react-native';
+import {useSelector} from 'react-redux';
+import {useAppDispatch} from '../redux/hook';
+import {clearError} from '../redux/actions';
+import Icon from 'react-native-vector-icons/MaterialIcons';
+import {RootState} from '../models/types';
+import {useNavigation} from '@react-navigation/native';
+import Colors from '../utils/Colors';
+
+interface ModalInterface {
+  modalVisible: boolean;
+  setModalVisible: Function;
+  page: string;
+  logo: string;
+}
+
+function ModalContainer({
+  modalVisible,
+  setModalVisible,
+  page,
+  logo,
+}: ModalInterface) {
+  const dispatch = useAppDispatch();
+  const error = useSelector(({Error}: RootState) => Error);
+  const navigation = useNavigation();
+
+  const handlePress = () => {
+    dispatch(clearError());
+
+    if (page == 'Detail') {
+      setModalVisible(false);
+      navigation.navigate('Home' as never);
```

Comment:
why as never? "Home" is a string, is strange you have to change it

Reply:
I used never because if not eslint was giving this error navigation.navigate('Home' as string); Argument of type 'string' is not assignable to parameter of type 'never'.